conformance/tests/httproute-partially-invalid-via-reference-grant.go

```diff
@@ -29,12 +29,12 @@ import (
 )
 
 func init() {
-	ConformanceTests = append(ConformanceTests, HTTPRouteInvalidReferenceGrant)
+	ConformanceTests = append(ConformanceTests, HTTPRoutePartiallyInvalidViaInvalidReferenceGrant)
 }
 
-var HTTPRouteInvalidReferenceGrant = suite.ConformanceTest{
-	ShortName:   "HTTPRouteInvalidReferenceGrant",
-	Description: "A single HTTPRoute in the gateway-conformance-infra namespace should fail to attach to a Gateway in the same namespace if the route has a backendRef Service in the gateway-conformance-app-backend namespace and a ReferenceGrant exists but does not grant permission to route to that specific Service",
+var HTTPRoutePartiallyInvalidViaInvalidReferenceGrant = suite.ConformanceTest{
+	ShortName:   "HTTPRoutePartiallyInvalidViaInvalidReferenceGrant",
+	Description: "A single HTTPRoute in the gateway-conformance-infra namespace should attach and be partially failed for a Gateway in the same namespace if the route has a backendRef Service in the gateway-conformance-app-backend namespace and a ReferenceGrant exists but does not grant permission to route to that specific Service",
```

Comment:
Tiny nit: "be partially failed" was a bit confusing and I don't think we need it, open to alternatives though.
```suggestion
	Description: "A single HTTPRoute in the gateway-conformance-infra namespace should attach to a Gateway in the same namespace if the route has a backendRef Service in the gateway-conformance-app-backend namespace and a ReferenceGrant exists but does not grant permission to route to that specific Service",
```

Reply:
Nah, that makes sense.